ecoservice_partner_account_company_automatic/partner.py

```diff
@@ -30,14 +30,12 @@ def create(self, cr, uid, vals, context=None):
         """ Inherit method to create the accounts automatically for partners without parent
         """
         result = super(partner, self).create(cr, uid, vals, context=context)
-        if 'parent_id' in vals:
-            if not vals['parent_id'] or (vals['parent_id'] and vals.get('is_company')):
-                ctx = dict(context)
-                if 'customer' in vals and vals['customer']:
-                    ctx['type'] = 'receivable'
-                    self.create_accounts(cr, uid, [result], context=ctx)
-                if 'supplier' in vals and vals['supplier']:
-                    ctx['type'] = 'payable'
-                    self.create_accounts(cr, uid, [result], context=ctx)
+        ctx = dict(context)
+        if 'customer' in vals and vals['customer']:
+            ctx['type'] = 'receivable'
+            self.create_accounts(cr, uid, [result], context=ctx)
+        if 'supplier' in vals and vals['supplier']:
```

Comment:
`if vals.get('supplier'):`

Reply:
also, if both are checked, `create_accounts` will be called twice, is this the desired effect?

Reply:
Looks like. Overall I don't wont to touch too much this crappy code :-)